src/main/java/lotto/domain/Lotto.java

```diff
@@ -1,87 +1,56 @@
 package lotto.domain;
 
+import lotto.generator.NumberGenerator;
+
 import java.util.*;
-import java.util.stream.Collectors;
 
 public class Lotto {
     private static final int LOTTO_MAX_SOCKET = 6;
-    private static final String LOTTO_WINNING_NUMBER_SPLIT_KEYWORD = ",";
 
-    private Set<LottoNumber> lottoNumbers;
+    private final Set<LottoNumber> lottoNumbers;
 
-    public static Lotto newAutomatic() {
-        return new Lotto();
+    public static Lotto automatic(NumberGenerator numberGenerator) {
+        return new Lotto(numberGenerator);
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return new Lotto(generatorLottoNumbers(numberGenerator));
```